### PR TITLE
Display drules with automatic priorities 

### PR DESCRIPTION
**Pull request by pastk**

### CODE REVIEW

File: src/libkomwm.py [hunk 1]
```diff
@@ -341,34 +337,60 @@ def dump_priorities(prio_range, path, maxzoom):
...
+                            dr_type, dr_auto_comment = dr_type_comment
+                            dr_zoom = dr_type + oid
+                            if dr_auto_comment is not None:
+                                dr_zoom = f'{dr_zoom}({dr_auto_comment})'
+                            dr_zoom += ' ' + prettify_zooms(visibilities[cl][dr_type_comment][oid], maxzoom)
+                            # Drules matching this prio_range and object_id and
+                            # - an auto priority dr_type match or
+                            # - any other non-auto dr_type suitable
+                            if (oid == object_id and (dr_type == auto_dr_type and dr_auto_comment == auto_comment or
+                                auto_dr_type is None and dr_auto_comment is None and
```
Comment (vng) [inline src/libkomwm.py]: How it should be interpreted?
```
A and B or C and D and ...
```
A and (B or C) and D
(A and B) or (C and D)
?

Please, put braces here and split this gigantic if on smaller conditions with intermediate vars.

File: src/libkomwm.py [hunk 2]
```diff
@@ -403,11 +431,18 @@ def get_drape_priority(cl, dr_type, object_id):
         ranges_to_check = (PRIO_BG_BY_SIZE, PRIO_BG_TOP, PRIO_FG)
     for r in ranges_to_check:
         if prio_id in prio_ranges[r]['priorities']:
-            return prio_ranges[r]['priorities'][prio_id] + prio_ranges[r]['base']
+            priority = prio_ranges[r]['priorities'][prio_id]
+            if auto_dr_type:
```
Comment (vng) [inline src/libkomwm.py]: Should be "is not None" ?

File: src/libkomwm.py [hunk 3]
```diff
@@ -283,6 +277,8 @@ def print_warning(msg):
 def store_visibility(cl, dr_type, object_id, zoom):
     if object_id == '::default':
         object_id = ''
+    if type(dr_type) is str:
+        dr_type = (dr_type, None)
```
Comment (vng) [inline src/libkomwm.py]: How it can be not a string?

Comment (pastk) [reply src/libkomwm.py]: it can be a tuple for cases when auto_comment is passed along

Comment (pastk) [reply src/libkomwm.py]: I've changed the func's signature to make it more explicit.

File: src/libkomwm.py (hunk 2 above)
Comment (pastk) [reply src/libkomwm.py]: fixed

File: src/libkomwm.py (hunk 1 above)
Comment (pastk) [reply src/libkomwm.py]: fixed

File: src/libkomwm.py [hunk 4]
```diff
@@ -341,37 +336,65 @@ def dump_priorities(prio_range, path, maxzoom):
...
+                            # Drules matching this prio_range and object_id and
+                            # - an auto priority dr_type match or
+                            # - any other non-auto dr_type suitable
+                            is_auto_dr_match = dr_type == auto_dr_type and dr_auto_comment == auto_comment
+                            is_not_auto_dr = auto_dr_type is None and dr_auto_comment is None
+                            is_suitable_for_range = (
+                                (prio_range == PRIO_OVERLAYS and dr_type in ('icon', 'caption', 'pathtext', 'shield')) or
+                                (prio_range in (PRIO_FG, PRIO_BG_TOP) and dr_type in ('line', 'area')) or
+                                (prio_range == PRIO_BG_BY_SIZE and dr_type == 'area'))
+                            if oid == object_id and (is_auto_dr_match or is_not_auto_dr and is_suitable_for_range):
```
Comment (vng) [inline src/libkomwm.py:394]: Again, (X or Y and Z) is undefined for me ..
Put braces here.
BTW, how python interpret this?

Comment (pastk) [reply src/libkomwm.py:394]: In all programming languages that I know `and` has a precedence over `or`.

**Pull request merged**